crates/op-rpc-types/src/op/transaction/receipt.rs

```diff
@@ -0,0 +1,196 @@
+use crate::op::transaction::tx_type;
+use alloy::{rpc::types::eth::Log, serde as alloy_serde};
+use alloy_consensus::{AnyReceiptEnvelope, ReceiptEnvelope, TxType};
+use alloy_primitives::{Address, B256};
+use serde::{Deserialize, Serialize};
+/// Transaction receipt
+///
+/// This type is generic over an inner [`ReceiptEnvelope`] which contains
+/// consensus data and metadata.
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+#[cfg_attr(
+    any(test, feature = "arbitrary"),
+    derive(proptest_derive::Arbitrary, arbitrary::Arbitrary)
+)]
+#[serde(rename_all = "camelCase")]
+pub struct TransactionReceipt<T = ReceiptEnvelope<Log>> {
```

Comment:
we need to solve this a bit differently
this is basically the ethereum receipt with extra l1 fee fields, but the receipt type is different, so we need a receipt envelope type that can also capture the extra op deposit fields

Reply:
True, I wanted this PR to be just for the RPC types and then make a separate PR for consensus. I'll just add a placeholder for now to not forget this.